src/libraries/System.Private.CoreLib/src/System/Threading/EventWaitHandle.Unix.cs

```diff
@@ -19,7 +19,7 @@ private void CreateEventCore(bool initialState, EventResetMode mode, string name
             createdNew = true;
         }
 
-        private static OpenExistingResult OpenExistingWorker(string name, out EventWaitHandle result)
+        private static OpenExistingResult OpenExistingWorker(string name, out EventWaitHandle? result)
```

Comment:
This might be better to move to the caller. It seems to do always do some error checking which is Windows specific

Reply:
I think we have to match the Windows equivalent here, since otherwise you end up with a bunch of weird platform-specific code at a higher level? Maybe I'm misunderstanding.

Reply:
I was referring to code patterns like https://github.com/dotnet/runtime/blob/1d9e50cb4735df46d3de0cee5791e97295eaf588/src/libraries/System.Private.CoreLib/src/System/Threading/EventWaitHandle.cs#L32-L45. The API is windows only and on Unix will have never reached error handling.